signal/micro/kernels/xtensa/fft_auto_scale.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "signal/src/fft_auto_scale.h"
+
+#include <math.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "signal/micro/kernels/fft_auto_scale.h"
+#include "signal/src/max_abs.h"
+#include "signal/src/msb.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/micro_context.h"
+
+#if XCHAL_HAVE_HIFI3
```

Comment:
We've typically used our own defines (HIFI3/4/5). Any reason to prefer the XCHAL versions?

Seems like it would simplify our life to just entirely switch over to the XCHAL defines, but I'm not sure what that might break.

Reply:
Following the source here. It does seem simpler this way.